src/queues/controllers/batch_processor.clj

```diff
@@ -0,0 +1,102 @@
+(ns queues.controllers.batch-processor
+  (:require [clojure.pprint :as pp]
+            [clojure.spec.alpha :as s]
+            [queues.cli :as cli]
+            [queues.init :as init]
+            [queues.io :as io]
+            [queues.json-converter :as json-converter]
+            [queues.logic.jobs :as logic.jobs]
+            [queues.specs.db :as specs.db]
+            [queues.specs.event-processor :as specs.event-processor]
+            [queues.specs.events :as specs.events]
+            [queues.specs.jobs-assigned :as specs.jobs-assigned]
+            [queues.state :as state]))
+
+(defn populate-db-applying-event-processor-on-batch-events!
+  [db event-processor batch-events]
+  (run! #(event-processor db %) batch-events))
+
+;; FIXME spec above failing when calling tests of batch-processor
+;;(s/fdef populate-db-applying-event-processor-on-batch-events!
+;;        :args (s/cat :db ::specs.db/db
+;;                     :event-processor ::specs.event-processor/event-processor
+;;                     :batch-events ::specs.events/events))
+
+(defn- all-job-assigned-events
+  [db]
+  ;; FIXME this structure does not guarantee that the jobs assigned will be delivered in the
+  ;; order they were assigned. If a job was assigned first but then finished later it will be
+  ;; displayed first in this vector
+  (let [all-jobs-with-assigned-agent (concat (state/all-jobs-done db)
```

Comment:
use (into ) instead of (concat )
here an explanation about why https://stuartsierra.com/2015/04/26/clojure-donts-concat

Reply:
muito interessante, obg!